### DISCUSSION TITLE
Feature Request: User Roles (profile settings, remove sidebar & 3-dot menu)

### DISCUSSION THREAD

**Discussion by YvesZH** [home-assistant/frontend#7614] [Configuration]
With the project stop of [custom-header](https://github.com/maykar/custom-header)  (@maykar thanks for your great work) I’m missing some functions in Home-Assistant.

- **Remove the menu sidebar for defined users**
- **Remove the 3-Dot menu for defined users**

In my opinion, most of the options in the _profile settings_ should also be able to be set by the “administrator” (depending on the authorization set, the user cannot change them).

When I think of the future of home assistants, it makes sense to expand with a central user administration. Something like this:

![fr_users_3](https://user-images.githubusercontent.com/64573990/98409420-1c1e2e80-2073-11eb-9774-2c17a88abab0.jpg)

**Comment by KTibow** [comment]
You can already remove the 3-dot menu with setting the user type to user instead of admin. You can also change the sidebar to a sidebar button in the profile page.

**Reply by iantrich** [reply]
Going to mark this is answered as @KTibow hit the nail on the head here.

**Reply by YvesZH** [reply]
Thanks. I know the option to hide the sidebar-menu. But i like to remove completly for the other normal users. The 3-dot option just remove the lovelace settings in there, but not the help option in there.
![0C6FCFAC-D3E9-4A03-B75E-C17D7897B05F](https://user-images.githubusercontent.com/64573990/98435658-0f322700-20d5-11eb-82ca-6ad91b479eaf.jpeg)

**Comment by KTibow** [comment]


**Reply by YvesZH** [reply]
Thank you - good to know.  The menu is only a small part of my Feature Request. The most important thing is that you can deactivate the sidebar-menu for defined users. And the way to do this would be to massively expand the user control (as patched together in my first picture). Change from device roles to user roles.

**Reply by KTibow** [reply]
Well then how are you supposed to deactivate kiosk mode if you can't open the sidebar?

**Reply by YvesZH** [reply]
The sidebar-menu could only be activated in the 'Users' group. This option should be deactivated for the Administrator groups.

The two groups (Administrator & Users) already exist (see: I haven't changed anything in the right column in the background of my first picture)

**Reply by KTibow** [reply]
Ended up closing it.

**Comment by maykar** [comment]
For what it's worth, I will have per user settings in [kiosk-mode](https://github.com/maykar/kiosk-mode) very soon, started working on it a few days ago. I'm also updating it to hide the menu button completely when sidebar is also hidden.

**Reply by KTibow** [reply]
kiosk-mode in native?

**Reply by maykar** [reply]
no the linked custom addon

**Reply by KTibow** [reply]
Do you want to add it to native sometime soon, or do you have other plans first?

**Reply by maykar** [reply]
To implement a kiosk mode in core presents quite a few challenges and even more questions/issues.

**Reply by maykar** [reply]
I was just offering an alternative and this is really more of a user settings/accessibility request than a Kiosk mode one.

**Reply by YvesZH** [reply]
That are great news.  Thank you @maykar  for your effort.  Exactly, that's what I personally need. 

Regarding implementation in the HA core: I think that a few basic decisions would have to be made first about how to deal with user rules.  Until that is settled, it probably doesn't make sense. The last few extensions can only be set per device (profile settings), and not per user.  It seems like things are going in a different direction than clearly controllable user roles.  And with this post I also wanted to stimulate this consideration.

**Reply by maykar** [reply]
`"Do you want to add it to native sometime soon, or do you have other plans first?"` @KTibow 

I'd love to add a kiosk mode to core at some point, but the implementation of it would have to make sense and have safeguards so that no one gets stuck without needed UI elements. If a kiosk mode was added to core, I imagine it would end up being a user preference per device like "always hide the sidebar".

**Reply by YvesZH** [reply]
> I imagine it would end up being a user preference per device like "always hide the sidebar". @maykar 

I totally agree with you. And that's the point: we (not just I) like to have it per user and not device.


To clarify to other readers:
![fr_overview](https://user-images.githubusercontent.com/64573990/98450948-91f0cb80-2141-11eb-8cf8-3d37f00f8b14.png)

**Comment by Palmdale95** [comment]
Hello guys. I am just humbling over this discussion, because I am looking for a solution to this. In my case I have neighbour who take care of my pet when I am not home and they just need some specific functions and that is what I would like to customize. So for them there is no need of sidebar and the information stated there. They just need on simple dashboard and that's it. ;-).
So now I am wondering if this idea came to a pleasant end?
